wai-extra/Network/Wai/Test.hs

```diff
@@ -5,6 +5,7 @@ module Network.Wai.Test
     ( -- * Session
       Session
     , runSession
+    , runSessionWith, ClientState(..), initState
```

Comment:
About `ClientState`: what i want to do is get the state out of the result of `runSessionWith`, parse it, and feed it to the backend handlers when talking to the database directly.  (I know that this style of writing test has its opponents, but I like them because it lets you check invariants in the backend without having to expose them via rest api.)

I think what would also work for me is just exporting the type, the record field selector, and `initState`, then I could read and write the cookies using `s { clientCookies = ... }` (that works without having the constructor, right?), but I am not sure that's any better for you.

Instead, I propose c3221d7a60968ef18e4d13ad03eb8d620a4980b6.